language/stdlib/modules/LibraAccount.move

```diff
@@ -1238,14 +1357,19 @@ module LibraAccount {
             sender_account.sequence_number < (MAX_U64 as u64),
             Errors::limit_exceeded(ESEQUENCE_NUMBER)
         );
+        assert(
+            sender_account.sequence_number == txn_sequence_number,
+            Errors::invalid_argument(ESEQUENCE_NUMBER)
```

Comment:
Can this actually be hit?

We should probably comment all of these asserts for which are possible to hit and which are for invariants

Reply:
I ran all the tests and nothing broke with this change. From the Move perspective there's no reason why an invalid `txn_sequence_number` could be passed in. My opinion is that if there's no way this can be hit, then `txn_sequence_number` shouldn't be an argument and we should directly increment the account's sequence number.